packages/extension-base/src/background/handlers/State.ts

```diff
@@ -509,10 +509,16 @@ export default class State {
     return provider.unsubscribe(request.type, request.method, request.subscriptionId);
   }
 
-  public async saveMetadata (meta: MetadataDef): Promise<void> {
+  public async saveMetadata ({ types, ...restMeta }: MetadataDef): Promise<void> {
+    type TypesType = ReturnType<Parameters<typeof localStorageStores.chainMetadata.update>[0]>[string]['types']
+
     await localStorageStores.chainMetadata.update((currentContent) => ({
       ...currentContent,
-      [meta.genesisHash]: meta
+      [restMeta.genesisHash]: {
+        ...restMeta,
+        // Type assertion, because "MetadataDef.types" can contain the CodecClass which should not appear here (and is not serializable anyway, so no use of it in local storage)
```

Comment:
In the previous commit I solved it with [this line](https://github.com/Cardinal-Cryptography/aleph-zero-signer/pull/104/commits/2a959c5ed708fec44ff2116fb74b3d631fb80c9c#diff-c8abca9e2cb46e31217ab8170901fa7fa56201b814e179150bfc5001853355d7L31), but I missed the fact that it'd strip all the contents of the object during the validation (so that the result is actually `{}` as requested). Appending `.passthrough()` would make it not strip it, but then it changes the type and makes it incompatible iwth `MetadataDef`. I then went back to _proper_ validation to make it actually work and instead hack the types in here, cause I don't have a better idea of how to satisfie the types.

Reply:
It's from a polkadot lib.